File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "RCJA";

        String[] arr = {"R", "T", "C", "F", "M", "J", "A", "N"};

        Map<String, Integer> count = new HashMap<>();
        for(int i = 0; i < arr.length; i++){
            count.put(arr[i], 0);
        }

        for(int i = 0; i < survey.length; i++){
            String a = Character.toString(survey[i].charAt(0));
            String b = Character.toString(survey[i].charAt(1));
            if(choices[i] == 4){
                continue;
            }else if(choices[i] > 4){
                count.replace(b, count.get(b) + choices[i] - 4);
            }else{
                count.replace(a, count.get(a) + 4 - choices[i]);
            }
        }

        if(count.get("R") < count.get("T")){
            answer = answer.replace("R", "T");
        }
        if(count.get("C") < count.get("F")){
            answer = answer.replace("C", "F");
        }
        if(count.get("J") < count.get("M")){
            answer = answer.replace("J", "M");
        }
        if(count.get("A") < count.get("N")){
            answer = answer.replace("A", "N");
        }
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var result = ""
    var dict = ["R" : 0, "T" : 0, "C" : 0, "F" : 0, "J" : 0, "M" : 0, "A" : 0, "N" : 0]
    let count = survey.count
    for i in 0..<count {
        var str = Array(survey[i])
        var a = str[0]
        var b = str[1]
        var num = choices[i]
        if num == 4 {
            continue
        } else if num > 4 {
            dict[String(b)]! += num - 4
        } else {
            dict[String(a)]! += 4 - num
        }
    }
    
    if dict["T"]! > dict["R"]! {
        result.append("T")
    } else {
        result.append("R")
    }
    if dict["F"]! > dict["C"]! {
        result.append("F")
    } else {
        result.append("C")
    }
    if dict["M"]! > dict["J"]! {
        result.append("M")
    } else {
        result.append("J")
    }
    if dict["N"]! > dict["A"]! {
        result.append("N")
    } else {
        result.append("A")
    }
    
    return result
}